### DISCUSSION TITLE
Whisper Configuration

### DISCUSSION THREAD

**Discussion by benevenuti15** [sepinf-inc/IPED#2320] [Q&A]
Dear all,

Good afternoon.

I have searched the available topics but could not find a detailed guide for configuring Whisper, specifically for audio transcription. Is there any tutorial or support material on how to set up this solution for transcribing audio files?

Although I am aware that this functionality will be included in the next version, version 4.2 does not yet have a defined release date, and I have an urgent need that needs to be addressed. Therefore, I would like to configure Whisper as soon as possible to meet this requirement.

Thank you for your assistance.

**Comment by gfd2020** [comment]
Will you use the GPU? 
You can now configure Python Modules on your computer.
On the wiki page there is a draft of the procedures.
https://github.com/sepinf-inc/IPED/wiki/User-Manual

Install the IPED python modules (also described in the WIKI):

whisperx
numpy
GPUtil


Another measure would be to download ffmpeg and place it in the path (in the case of WhisperX)

**Comment by lfcnassif** [comment]
Download the last 4.2 snapshot:
https://github.com/sepinf-inc/IPED/actions/runs/10710122355

The specific wiki section is this:
https://github.com/sepinf-inc/IPED/wiki/User-Manual#whisper

After installing ONE of the recommended libraries (I recommend faster-whisper if you will run on CPU), go to conf/AudioTranscriptConfig.txt and uncomment the line below:
`implementationClass = iped.engine.task.transcript.WhisperTranscriptTask`

On `Local WhisperTranscriptTask options` section there are some optional configurations, like the model size.

`language = auto` uses the language specified in LocalConfig.txt for all audios.
`language = detect` uses automatic language detection for each audio.

> whisperx

The original library doesn't compute confidence scores, we recommend our fork pointed out in the wiki (just if whisperx is used).

**Reply by benevenuti15** [reply]
Thank you @gfd2020  and @lfcnassif  for the clarifications. 
However, I am having difficulty in the process of running portable python for Windows. Should I use the executable found in the \IPED-4.1.6\iped-4.1.6\python folder to install packages with pip?

**Reply by gfd2020** [reply]
Yes.
In the path of your iped installation, run the commands on CMD, example ( c:\IPED-4.1.6\iped-4.1.6>   ) .

.\python\python.exe .\python\get-pip.py

.\python\Scripts\pip.exe install numpy GPUtil faster_whisper

**Reply by benevenuti15** [reply]
Thanks, @gfd2020 
Download the packages now!
